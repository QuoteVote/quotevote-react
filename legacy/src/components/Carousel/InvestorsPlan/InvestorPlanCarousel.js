import React, { useState } from 'react'
import { CircularProgress, TextField, Typography } from '@material-ui/core'
import Button from '@material-ui/core/Button'
import PropTypes from 'prop-types'
import InputAdornment from '@material-ui/core/InputAdornment'
import { useHistory } from 'react-router-dom'
import { isMobile } from 'react-device-detect'
import { useMutation } from '@apollo/react-hooks'
import { useForm } from 'react-hook-form'
import Carousel from 'react-material-ui-carousel'
import Grid from '@material-ui/core/Grid'
import withWidth from '@material-ui/core/withWidth'
import Hidden from '@material-ui/core/Hidden'
import GridItem from '../../../mui-pro/Grid/GridItem'
import InvestorContent1Image from '../../../assets/img/InvestorContent1.png'
import InvestorContent2Image from '../../../assets/svg/InvestorContent2.svg'
import InvestorContent3Image from '../../../assets/svg/InvestorContent3.svg'
import InvestButton from '../../CustomButtons/InvestButton'
import { SEND_INVESTOR_EMAIL } from '../../../graphql/mutations'
import DoubleArrowIconButton from '../../CustomButtons/DoubleArrowIconButton'
import { MOBILE_IMAGE_WIDTH } from '../../../views/LandingPage/LandingPage'

InvestorCarouselFirstContent.propTypes = {
  classes: PropTypes.object,
  setContentIndex: PropTypes.func,
  width: PropTypes.string,
}

function InvestorCarouselFirstContent({ classes, setContentIndex, width }) {
  const { opinionsText, bottomText, greenText } = classes
  const browserWidth = '400.43px'
  const imageWidth = isMobile ? MOBILE_IMAGE_WIDTH : browserWidth
  return (
    <Grid
      container
      direction="row"
      justify="center"
      alignItems="center"
      style={{
        paddingLeft: width === 'xs' ? 20 : 60,
        paddingRight: width === 'xs' ? 0 : 60,
      }}
    >
      <Grid item xs={12} sm={12} md={6}>
        <img
          alt="Investor"
          src={InvestorContent1Image}
          style={{
            width: imageWidth,
            height: isMobile ? 'auto' : '350.51px',
            objectFit: 'contain',
          }}
        />
      </Grid>
      <GridItem xs={12} sm={12} md={6}>
        <Typography>
          <div className={opinionsText}>
            We created VoxPop with promoting
            {' '}
            <b>democracy and community as our pillar.</b>
            <br />
            <br />
            This is why we reject investment from VC firms, and encourage users to invest and become a part of the
            change.
            <b>Invest up to $2000 to grow with us. </b>
            <br />
            <br />
          </div>
        </Typography>
        <Hidden mdUp>
          <InvestButton
            width={width}
            handleClick={() => {
              setContentIndex(2)
            }}
          />
        </Hidden>
        <Typography className={bottomText}>
          <Hidden smDown>
            <InvestButton
              width={width}
              handleClick={() => {
                setContentIndex(2)
              }}
            />
          </Hidden>
          {' '}
          What is
          <span className={greenText}> the deal </span>
          <DoubleArrowIconButton onClick={() => setContentIndex(1)} />
        </Typography>
      </GridItem>
    </Grid>
  )
}

InvestorCarouselSecondContent.propTypes = {
  classes: PropTypes.object,
  setContentIndex: PropTypes.func,
  width: PropTypes.string,
}

function InvestorCarouselSecondContent({ classes, setContentIndex, width }) {
  const { opinionsText, bottomText, greenText } = classes
  const browserWidth = '435.43px'
  const imageWidth = isMobile ? MOBILE_IMAGE_WIDTH : browserWidth
  return (
    <Grid
      container
      direction="row"
      justify="center"
      alignItems="center"
      style={{
        paddingLeft: width === 'xs' ? 20 : 60,
        paddingRight: width === 'xs' ? 0 : 60,
      }}
    >
      <Grid item xs={12} sm={12} md={7} lg={6}>
        <img
          alt="Investor 2"
          src={InvestorContent2Image}
          style={{
            width: imageWidth,
            height: isMobile ? 'auto' : '350.51px',
            objectFit: 'contain',
          }}
        />
      </Grid>
      <Grid item xs={12} sm={12} md={5} lg={6}>
        <Typography>
          <div className={opinionsText}>
            By capping individual investments at $2000,
            {' '}
            <b>we ensure shareholders have an equal voice, and provide a more inclusive opportunity to invest. </b>
            <br />
            <br />
            We keep growth organized and open through voxPOP and
            {' '}
            <b>all shareholder opinions hold the same weight. </b>
            <br />
            <br />
          </div>
        </Typography>
        <Hidden mdUp>
          <InvestButton
            width={width}
            handleClick={() => {
              setContentIndex(2)
            }}
          />
        </Hidden>
        <Typography className={bottomText}>
          <Hidden smDown>
            <InvestButton
              width={width}
              handleClick={() => {
                setContentIndex(2)
              }}
            />
          </Hidden>
          {'  '}
          I want to
          {' '}
          <span className={greenText}> to know details</span>
          <DoubleArrowIconButton onClick={() => setContentIndex(2)} />
        </Typography>
      </Grid>
    </Grid>
  )
}

InvestorCarouselThirdContent.propTypes = {
  classes: PropTypes.object,
  width: PropTypes.object,
}

function InvestorCarouselThirdContent({ classes, width }) {
  const history = useHistory()
  const [sendInvestorMail, { data, error, loading }] = useMutation(SEND_INVESTOR_EMAIL)
  const {
    register, errors, handleSubmit, getValues,
  } = useForm()

  const handleSendEmail = async () => {
    const { email } = getValues()
    await sendInvestorMail({ variables: { email } })
  }
  const browserWidth = width === 'md' ? '400.43px' : '435.43px'
  const imageWidth = isMobile ? MOBILE_IMAGE_WIDTH : browserWidth

  if (data) {
    history.push('/auth/investor-thanks')
  }

  return (
    <form onSubmit={handleSubmit(handleSendEmail)}>
      <Grid
        container
        direction="row"
        justify="center"
        alignItems="center"
        style={{
          paddingLeft: width === 'xs' ? 20 : 60,
          paddingRight: width === 'xs' ? 0 : 60,
        }}
      >
        <Grid item xs={12} sm={12} md={8} lg={7}>
          <img
            alt="Investor"
            src={`${InvestorContent3Image}`}
            style={{
              width: imageWidth,
              height: isMobile ? 'auto' : '350.51px',
              objectFit: 'contain',
            }}
          />
        </Grid>
        <Grid item xs={12} sm={12} md={4} lg={5}>
          <Typography>
            <div className={classes.opinionsText}>
              <b>There are 10,000,000 shares of stock</b>
              <br />
              Join us in creating a truly open and equal community
              where civil conversation is the main objective.
              <br />
              <br />
              <br />
            </div>
          </Typography>
          <div className={classes.sendEmail}>
            <TextField
              variant="filled"
              required
              label="Email"
              name="email"
              id="email"
              error={errors.email}
              helperText={errors.email && errors.email.message}
              fullWidth
              inputRef={register({
                required: 'Email is required',
                pattern: {
                  value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
                  message: 'Invalid email address',
                },
              })}
              InputProps={{
                endAdornment: (
                  <InputAdornment position="end">
                    <Button
                      aria-label="Send"
                      className={classes.sendEmailButton}
                      onClick={handleSendEmail}
                      disabled={loading || error || errors.email}
                    >
                      Send
                      {loading && (<CircularProgress size={20} className={classes.loadingProgress} />)}
                    </Button>
                  </InputAdornment>
                ),
              }}
            />
          </div>
        </Grid>
      </Grid>
    </form>
  )
}

function InvestorPlanCarousel(props) {
  const [contentIndex, setContentIndex] = useState(0)
  const { setCarouselCurrentIndex, classes } = props
  return (
    <Carousel
      navButtonsAlwaysVisible
      index={contentIndex}
      onChange={(index) => {
        setCarouselCurrentIndex(index)
      }}
      activeIndicatorProps={{
        className: classes.activeIndicator,
      }}
      indicatorProps={{
        className: classes.inactiveIndicator,
      }}
    >
      <InvestorCarouselFirstContent {...props} setContentIndex={setContentIndex} />
      <InvestorCarouselSecondContent {...props} setContentIndex={setContentIndex} />
      <InvestorCarouselThirdContent {...props} />
    </Carousel>
  )
}

InvestorPlanCarousel.propTypes = {
  classes: PropTypes.object,
  setCarouselCurrentIndex: PropTypes.func,
}

export default withWidth()(InvestorPlanCarousel)
