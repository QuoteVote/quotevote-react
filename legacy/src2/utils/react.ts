import React, { 
  useState, 
  useEffect, 
  useRef, 
  ReactNode, 
  ChangeEvent, 
  KeyboardEvent,
  FormEvent,
  MouseEvent,
  Dispatch,
  SetStateAction
} from 'react';

export type { 
  ReactNode,
  ChangeEvent,
  KeyboardEvent,
  FormEvent,
  MouseEvent,
  Dispatch,
  SetStateAction
};

export { 
  React as default,
  useState,
  useEffect,
  useRef
};

// Add type declarations for common event handlers
export type InputChangeEvent = ChangeEvent<HTMLInputElement>;
export type TextAreaChangeEvent = ChangeEvent<HTMLTextAreaElement>;
export type ButtonClickEvent = MouseEvent<HTMLButtonElement>;
export type SetState<T> = Dispatch<SetStateAction<T>>;

// Add type declarations for common component props
export interface BaseProps {
  className?: string;
  style?: React.CSSProperties;
  children?: ReactNode;
}

export interface InputProps extends BaseProps {
  value: string;
  onChange: (value: string) => void;
  placeholder?: string;
  disabled?: boolean;
} 