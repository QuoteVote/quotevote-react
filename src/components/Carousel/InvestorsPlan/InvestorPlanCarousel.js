import React, { useState } from 'react'
import { CircularProgress, TextField, Typography } from '@material-ui/core'
import Button from '@material-ui/core/Button'
import PropTypes from 'prop-types'
import InputAdornment from '@material-ui/core/InputAdornment'
import { useHistory } from 'react-router-dom'
import { useMutation } from '@apollo/react-hooks'
import { useForm } from 'react-hook-form'
import GridItem from '../../../mui-pro/Grid/GridItem'
import GridContainer from '../../../mui-pro/Grid/GridContainer'
import investorPlanImg from '../../../assets/img/UserSharing.png'
import investorPlanImg2 from '../../../assets/img/CommentBox2.png'
import investorPlanImg3 from '../../../assets/img/GroupChat.png'
import InvestButton from '../../InvestButton'
import Carousel from '../Carousel'
import { SEND_INVESTOR_EMAIL } from '../../../graphql/mutations'
import DoubleArrowIconButton from '../../DoubleArrowIconButton'

InvestorCarouselFirstContent.propTypes = {
  classes: PropTypes.object,
  handleNext: PropTypes.func,
  setActiveStepProp: PropTypes.func,
}

function InvestorCarouselFirstContent({ classes, handleNext, setActiveStepProp }) {
  const { opinionsText, bottomText, greenText } = classes
  return (
    <GridContainer justify="center" style={{ marginRight: 24 }}>
      <GridItem xs={12} sm={5}>
        <GridContainer justify="center">
          <img
            alt={investorPlanImg}
            src={`${investorPlanImg}`}
            style={{
              width: '400.43px',
              height: '300.51px',
              objectFit: 'contain',
            }}
          />
        </GridContainer>
      </GridItem>
      <GridItem xs={12} sm={4}>
        <GridContainer justify="left">
          <Typography>
            <div className={opinionsText}>
              We created VoxPop with promoting
              {' '}
              <b>democracy and community as our pillar.</b>
              <br />
              <br />
              This is why we reject investment from VC firms, and encourage users to invest and become a part of the
              change.
              <b>Invest up to $2000 to grow with us. </b>
              <br />
              <br />
            </div>
          </Typography>
          <Typography className={bottomText}>
            <InvestButton
              handleClick={() => {
                setActiveStepProp(2)
              }}
            />
            {' '}
            What is
            <span className={greenText}> the deal </span>
            <DoubleArrowIconButton onClick={() => handleNext(1)} />
          </Typography>
        </GridContainer>
      </GridItem>
    </GridContainer>
  )
}

InvestorCarouselSecondContent.propTypes = {
  classes: PropTypes.object,
  handleNext: PropTypes.func,
  setActiveStepProp: PropTypes.func,
}

function InvestorCarouselSecondContent({ classes, handleNext, setActiveStepProp }) {
  const { opinionsText, bottomText, greenText } = classes
  return (
    <GridContainer justify="center" style={{ marginRight: 24 }}>
      <GridItem xs={12} sm={5}>
        <GridContainer justify="center">
          <img
            alt={investorPlanImg2}
            height={500}
            src={`${investorPlanImg2}`}
            style={{
              width: '435.43px',
              height: '300.51px',
              objectFit: 'contain',
            }}
          />
        </GridContainer>
      </GridItem>
      <GridItem xs={12} sm={4}>
        <GridContainer justify="left">
          <Typography>
            <div className={opinionsText}>
              By capping individual investments at $2000,
              {' '}
              <b>we ensure shareholders have an equal voice, and provide a more inclusive opportunity to invest. </b>
              <br />
              <br />
              We keep growth organized and open through voxPOP and
              {' '}
              <b>all shareholder opinions hold the same weight. </b>
              <br />
              <br />
            </div>
          </Typography>
          <Typography className={bottomText}>
            <InvestButton
              handleClick={() => {
                setActiveStepProp(2)
              }}
            />
            {'  '}
            I want to
            {' '}
            <span className={greenText}> to know details</span>
            <DoubleArrowIconButton onClick={() => handleNext(2)} />
          </Typography>
        </GridContainer>
      </GridItem>
    </GridContainer>
  )
}

InvestorCarouselThirdContent.propTypes = {
  classes: PropTypes.object,
}

function InvestorCarouselThirdContent({ classes }) {
  const history = useHistory()
  const [sendInvestorMail, { data, error, loading }] = useMutation(SEND_INVESTOR_EMAIL)
  const {
    register, errors, handleSubmit, getValues,
  } = useForm()

  const handleSendEmail = async () => {
    const { email } = getValues()
    await sendInvestorMail({ variables: { email } })
  }

  if (data) {
    history.push('/auth/investor-thanks')
  }

  return (
    <form onSubmit={handleSubmit(handleSendEmail)}>
      <GridContainer justify="center" style={{ marginRight: 24 }}>
        <GridItem xs={12} sm={5}>
          <GridContainer justify="center">
            <img
              alt={investorPlanImg3}
              height={500}
              src={`${investorPlanImg3}`}
              style={{
                width: '435.43px',
                height: '300.51px',
                objectFit: 'contain',
              }}
            />
          </GridContainer>
        </GridItem>
        <GridItem xs={12} sm={4}>
          <GridContainer justify="left">
            <Typography>
              <div className={classes.opinionsText}>
                <b>There are 10,000,000 shares of stock</b>
                <br />
                Join us in creating a truly open and equal community
                where civil conversation is the main objective.
                <br />
                <br />
                <br />
              </div>
            </Typography>
            <div className={classes.sendEmail}>
              <TextField
                variant="filled"
                required
                label="Email"
                name="email"
                id="email"
                error={errors.email}
                helperText={errors.email && errors.email.message}
                inputRef={register({
                  required: 'Email is required',
                  pattern: {
                    value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
                    message: 'Invalid email address',
                  },
                })}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end">
                      <Button
                        aria-label="Send"
                        className={classes.sendEmailButton}
                        onClick={handleSendEmail}
                        disabled={loading || error || errors.email}
                      >
                        Send
                        {loading && (<CircularProgress size={20} className={classes.loadingProgress} />)}
                      </Button>
                    </InputAdornment>
                  ),
                }}
              />
            </div>
          </GridContainer>
        </GridItem>
      </GridContainer>
    </form>
  )
}

function InvestorPlanCarousel(props) {
  const [activeStepProp, setActiveStepProp] = useState(0)
  const handleNext = () => {
    setActiveStepProp((prevActiveStep) => prevActiveStep + 1)
  }

  return (
    <Carousel
      autoplay={false}
      navButtonsAlwaysVisible={false}
      activeStepProp={activeStepProp}
      setActiveStepProp={setActiveStepProp}
    >
      <InvestorCarouselFirstContent {...props} handleNext={handleNext} setActiveStepProp={setActiveStepProp} />
      <InvestorCarouselSecondContent {...props} handleNext={handleNext} setActiveStepProp={setActiveStepProp} />
      <InvestorCarouselThirdContent {...props} />
    </Carousel>
  )
}

InvestorPlanCarousel.propTypes = {
  classes: PropTypes.object,
}

export default InvestorPlanCarousel
