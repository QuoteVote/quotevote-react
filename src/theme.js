import orange from '@material-ui/core/colors/orange'
// import blue from '@material-ui/core/colors/blue'
// import grey from '@material-ui/core/colors/grey'
import pink from '@material-ui/core/colors/pink'
import red from '@material-ui/core/colors/red'
import green from '@material-ui/core/colors/green'
// import yellow from '@material-ui/core/colors/yellow'
import cyan from '@material-ui/core/colors/cyan'
import teal from '@material-ui/core/colors/teal'

const theme = {
  activityCards: {
    quoted: {
      color: cyan[500],
    },
    commented: {
      color: orange[500],
    },
    upvote: {
      color: green[500],
    },
    downvote: {
      color: red[500],
    },
    submitted: {
      color: '#000000',
    },
    hearted: {
      color: pink,
    },
  },
  subHeader: {
    activeIcon: {
      color: teal.A400,
    },
    default: {
      color: 'black',
    },
    followButton: {
      backgroundColor: '#00CF6E',
      color: 'white',
    },
  },
}

export default theme
