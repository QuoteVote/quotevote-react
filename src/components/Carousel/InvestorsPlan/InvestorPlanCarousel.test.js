import React, { useState } from 'react'
import { render } from '@testing-library/react'

import { makeStyles, MuiThemeProvider as ThemeProvider } from '@material-ui/core/styles'
import InvestorPlanCarousel from './InvestorPlanCarousel'
import styles from '../../../assets/jss/material-dashboard-pro-react/views/landingPageStyle'
import theme from '../../../themes/MainTheme'

const useStyles = makeStyles(styles)

function InvestorPlanCarouselWrapper() {
  const classes = useStyles()
  const [carouselCurrentIndex, setCarouselCurrentIndex] = useState(0)
  return (

    <ThemeProvider theme={theme}>
      <InvestorPlanCarousel classes={classes} setCarouselCurrentIndex={setCarouselCurrentIndex} carouselCurrentIndex={carouselCurrentIndex} />
    </ThemeProvider>
  )
}

describe('InvestorPlanCarousel test -', () => {
  it('renders correctly', () => {
    const { container } = render(<InvestorPlanCarouselWrapper />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
