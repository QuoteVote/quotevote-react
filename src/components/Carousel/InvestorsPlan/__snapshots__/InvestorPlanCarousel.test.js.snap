// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvestorPlanCarousel test - renders correctly 1`] = `null`;
